web/src/app/pages/main-page-container/main-page/secondary-side-panel/submission-panel/submission-panel.component.ts

```diff
@@ -107,22 +107,30 @@ export class SubmissionPanelComponent implements OnInit, OnDestroy {
     return this.getTaskSubmissionResult(task)!.value as MultipleSelection;
   }
 
+  getTaskMultipleChoiceOtherValue(task: Task): string | null {
+    const multipleSelection = this.getTaskSubmissionResult(task)!
+      .value as MultipleSelection;
+    // Ensure at least one value is present: if no values are selected and 'otherText' is empty, add 'Other' as a fallback.
+    if (multipleSelection.values.size === 0 && !multipleSelection.otherValue)
+      return 'Other';
+    if (multipleSelection.otherValue)
```

Comment:
Please refactor duplicate logic into shared lib.

Reply:
in another pr :)